packages/cascara/src/structures/AdminStructure/components/Nav/Nav.js

```diff
@@ -14,21 +15,22 @@ const sectionShape = {
 };
 
 const propTypes = {
+  as: pt.string,
   links: pt.arrayOf(
     pt.oneOfType([pt.shape(LINK_SHAPE), pt.shape(sectionShape)])
   ),
 };
 
-const Nav = ({ links, ...rest }) => (
-  <div {...rest} className={cx('Nav', rest.className)}>
+const Nav = ({ as, links, ...rest }) => (
+  <Role {...rest} as={as} className={cx('Nav', rest.className)}>
```

Comment:
@rociobaena this component should be the reference point for all other components in this PR as a pattern. Everything here is correct and other components are missing parts of this:

- Rest spread is first
- `className` is not part of propTypes and is instead coming from `rest.className`
- The component's CSS Module class is defined first, then the `rest.className` is defined last
- There is a default prop assigned for `as`

All other components that are using `Role` need to have all of these things.